### PR TITLE
Regression test issuance

### PR DESCRIPTION
**Pull request by bachlx-bitmark**
there is only photo issuance flow.

### CODE REVIEW

File: __tests__/common/common.js [hunk 1]
```diff
@@ -130,9 +132,79 @@ const closeWhatNewScreen = async (driver) => {
...
+        // Choose image from lib
+        .waitForElementByName('Choose from Library...', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('Choose from Library...').tap();
+    try {
+        // allow permission
+        await driver.waitForElementByName('OK', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('OK').tap();
+    } catch {
+        console.log('Already allowed permission before');
+    }
+
+    // .waitForElementByName('OK', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('OK').tap()
```
Comment (phieudu241) [inline __tests__/common/common.js]: Should remove this commented line

File: __tests__/common/common.js [hunk 2]
```diff
@@ -130,9 +132,79 @@ const closeWhatNewScreen = async (driver) => {
...
+    let elements = driver.waitForElementByName('Camera Roll', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('Camera Roll').tap()
+        // Choose image from lib
+        .waitForElementsByIosPredicateString("type == 'XCUIElementTypeCell'", TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementsByIosPredicateString("type == 'XCUIElementTypeCell'");
+    // Choose latest image
+    await elements[elements.length - 1].tap();
+
+    let textInputAssetName = await driver.waitForElementById('inputAssetName', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).waitForElementById('inputAssetName');
```
Comment (phieudu241) [inline __tests__/common/common.js]: Duplication: `waitForElementById('inputAssetName')`

File: __tests__/common/common.js [hunk 3]
```diff
@@ -130,9 +132,79 @@ const closeWhatNewScreen = async (driver) => {
...
+    let textInputQuantity = await driver.waitForElementById('inputQuantity', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).waitForElementById('inputQuantity');
```
Comment (phieudu241) [inline __tests__/common/common.js]: Duplication `.waitForElementById('inputQuantity')`

File: __tests__/common/common.js [hunk 4]
```diff
@@ -130,9 +132,79 @@ const closeWhatNewScreen = async (driver) => {
...
+
+    await textInputAssetName.type(assetName);
+    await driver.hideKeyboard();
+    let length = (await driver.elementsById('errorInputAssetName')).length;
```
Comment (phieudu241) [inline __tests__/common/common.js]: should rename variable: `length` to `numberOfErrors`

File: __tests__/common/common.js [hunk 5]
```diff
@@ -130,9 +132,79 @@ const closeWhatNewScreen = async (driver) => {
...
+
+    // quantity
+    await textInputQuantity.clear().type(quantity);
+    await driver.hideKeyboard({ strategy: 'pressKey', key: 'Done' });
+    length = (await driver.elementsById('errorInputQuantity')).length;
+    expect(length).toEqual(0);
+
+    await driver.waitForElementByName('ISSUE', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('ISSUE').tap();
+    await driver.sleep(20 * 1000);
+    await driver.waitForElementByName('OK', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).waitForElementByName('OK').tap();
```
Comment (phieudu241) [inline __tests__/common/common.js]: Duplication: `.waitForElementByName('OK')`

File: src/utils/file-util.js [hunk 6]
```diff
@@ -3,6 +3,8 @@ import RNFS from 'react-native-fs';
 import { zip, unzip } from 'react-native-zip-archive';
 import { config } from 'src/configs';
 
+console.log('RNFS :', RNFS);
```
Comment (phieudu241) [inline src/utils/file-util.js]: Should remove this log

File: src/views/components/home/bottom-tabs/bottom-tabs.component.js [hunk 7]
```diff
@@ -30,10 +30,13 @@ export class PrivateBottomTabsComponent extends BottomTabBar {
...
-            return (<OneTabButtonComponent key={index} style={componentStyles.bottomTabButton} onPress={() => {
-              this.props.onTabPress({ route });
-              Actions.reset('properties');
-            }}>
+            return (<OneTabButtonComponent key={index} style={componentStyles.bottomTabButton}
+              testID={`BottomTabsComponent_${label}`}
+              onPress={() => {
+                this.props.onTabPress({ route });
+                Actions.reset('properties');
+
```
Comment (phieudu241) [inline src/views/components/home/bottom-tabs/bottom-tabs.component.js]: Should remove this blank line

File: src/views/components/home/properties/local-issue-file/issue-file/issue-file.component.js [hunk 8]
```diff
@@ -233,15 +234,19 @@ export class LocalIssueFileComponent extends React.Component {
                   data={this.state.metadataList}
                   extraData={this.state}
                   renderItem={({ item }) => {
+                    console.log('btnRemoveMetadataLabel_' + item.key);
```
Comment (phieudu241) [inline src/views/components/home/properties/local-issue-file/issue-file/issue-file.component.js]: Should remove this log

File: __tests__/common/common.js [hunk 9]
```diff
@@ -130,9 +132,79 @@ const closeWhatNewScreen = async (driver) => {
...
+};
```
Comment (phieudu241) [inline __tests__/common/common.js]: Need to verify result after issuing

Comment (bachlx-bitmark) [reply __tests__/common/common.js]: check result after issuing in file photo.test.js

File: src/views/components/home/properties/local-issue-file/issue-file/issue-file.component.js (hunk 8 above)
Comment (bachlx-bitmark) [reply src/views/components/home/properties/local-issue-file/issue-file/issue-file.component.js]: done

File: src/views/components/home/bottom-tabs/bottom-tabs.component.js (hunk 7 above)
Comment (bachlx-bitmark) [reply src/views/components/home/bottom-tabs/bottom-tabs.component.js]: done

File: __tests__/common/common.js (hunk 5 above)
Comment (bachlx-bitmark) [reply __tests__/common/common.js]: done

File: __tests__/common/common.js (hunk 3 above)
Comment (bachlx-bitmark) [reply __tests__/common/common.js]: done

File: __tests__/common/common.js (hunk 4 above)
Comment (bachlx-bitmark) [reply __tests__/common/common.js]: done

File: __tests__/common/common.js (hunk 1 above)
Comment (bachlx-bitmark) [reply __tests__/common/common.js]: done

File: __tests__/common/common.js (hunk 2 above)
Comment (bachlx-bitmark) [reply __tests__/common/common.js]: done

File: __tests__/flows/issuance/photo.test.js [hunk 10]
```diff
@@ -0,0 +1,207 @@
+import wd from 'wd';
+import { APPIUM_CONFIG, RUN_CONFIG, TEST_CONFIG } from '../../configs/config'
+import { issueNewPhotoWithoutMetadata, pushNewPhotoToDevice, createNewAccountWithoutTouchId } from '__tests__/common/common';
+
+let path = require('path');
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = TEST_CONFIG.DEFAULT_TIMEOUT_INTERVAL;
+const driver = wd.promiseChainRemote(APPIUM_CONFIG.HOST, APPIUM_CONFIG.PORT);
+
+// beforeAll(async () => {
```
Comment (phieudu241) [inline __tests__/flows/issuance/photo.test.js]: Should remove these comment line

File: __tests__/flows/issuance/photo.test.js [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+  await textInputQuantity.clear();
+  await driver.hideKeyboard({ strategy: 'pressKey', key: 'Done' });
+  errorString = await driver.waitForElementById('errorInputQuantity', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('errorInputQuantity').text();
+  expect(errorString).toEqual('Number of bitmarks should be an integer number');
+  // quantity < 0
+  await textInputQuantity.type(-1);
+  await driver.hideKeyboard({ strategy: 'pressKey', key: 'Done' });
+  errorString = await driver.waitForElementById('errorInputQuantity', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('errorInputQuantity').text();
+  expect(errorString).toEqual('Create property requires a minimum quantity of 1 bitmark issuance.');
+  // quantity 100
```
Comment (phieudu241) [inline __tests__/flows/issuance/photo.test.js]: As below line: `type(5)` however the comment is `quality 100`

File: __tests__/flows/issuance/photo.test.js [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+  await textInputQuantity.clear().type(5);
+  await driver.hideKeyboard({ strategy: 'pressKey', key: 'Done' });
+  numberOfErrors = (await driver.elementsById('errorInputQuantity')).length;
+  expect(numberOfErrors).toEqual(0);
+  // add metadata 0
```
Comment (phieudu241) [inline __tests__/flows/issuance/photo.test.js]: What is `metadata 0`?

File: __tests__/flows/issuance/photo.test.js [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+  await btnAddMoreMetadata.tap();
+
+  await driver.waitForElementById('btnMetadataLabel_0', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('btnMetadataLabel_0').tap()
+    .waitForElementByName('CREATOR', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('CREATOR').tap();
+  await driver.sleep(2000);
+  let metadataLabel0 = await driver.waitForElementById('btnMetadataLabel_0', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('btnMetadataLabel_0').text();
+  expect(metadataLabel0).toEqual('CREATOR');
+  let textInputMetadataValue0 = await driver.elementById('inputMetadataValue_0');
+  // // input over 2048 chars
```
Comment (phieudu241) [inline __tests__/flows/issuance/photo.test.js]: Should remove these comment line

File: __tests__/flows/issuance/photo.test.js (hunk 12 above)
Comment (bachlx-bitmark) [reply __tests__/flows/issuance/photo.test.js]: updated comment, add first metadat

File: __tests__/flows/issuance/photo.test.js (hunk 11 above)
Comment (bachlx-bitmark) [reply __tests__/flows/issuance/photo.test.js]: updated type 100

File: __tests__/flows/issuance/photo.test.js [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+  // add metadata 1
+  await btnAddMoreMetadata.tap();
+  await driver.waitForElementById('btnMetadataLabel_1').elementById('btnMetadataLabel_1').tap()
+    .waitForElementByName('CREATOR', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('CREATOR').tap();
+  await driver.sleep(2000);
+  let metadataLabel1 = await driver.waitForElementById('btnMetadataLabel_1', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('btnMetadataLabel_1').text();
+  expect(metadataLabel1).toEqual('CREATOR');
+  errorString = await driver.waitForElementById('errorInputMetadata', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('errorInputMetadata').text();
+  expect(errorString).toEqual('Duplicated labels: creator');
+  await driver.waitForElementByName('EDIT', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('EDIT').tap()
```
Comment (phieudu241) [inline __tests__/flows/issuance/photo.test.js]: Should add a comment here to explain what these actions are for

File: __tests__/flows/issuance/photo.test.js (hunk 10 above)
Comment (bachlx-bitmark) [reply __tests__/flows/issuance/photo.test.js]: done

File: __tests__/flows/issuance/photo.test.js [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+
+  let elements = await driver
+    .waitForElementById('BottomTabsComponent_properties', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('BottomTabsComponent_properties').tap()
+    .waitForElementById('addPropertyBtn', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementById('addPropertyBtn').tap()
+    // select photo
+    .waitForElementByName('PHOTOS', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('PHOTOS').tap()
+    // Choose image from lib
+    .waitForElementByName('Choose from Library...', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('Choose from Library...').tap()
+    // allow permission
+    // .waitForElementByName('OK', TEST_CONFIG.CHANGE_SCREEN_TIMEOUT).elementByName('OK').tap()
```
Comment (phieudu241) [inline __tests__/flows/issuance/photo.test.js]: Should remove this log

File: __tests__/flows/issuance/photo.test.js (hunk 14 above)
Comment (bachlx-bitmark) [reply __tests__/flows/issuance/photo.test.js]: done

File: __tests__/flows/issuance/photo.test.js (hunk 15 above)
Comment (bachlx-bitmark) [reply __tests__/flows/issuance/photo.test.js]: done

**Pull request merged**